app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Api::V1::BookmarksController < Api::V1Controller
+  before_action :set_bookmark, only: [:show, :update, :destroy]
+
+  # GET /api/v1/bookmarks
+  def index
+    @bookmarks = current_user.bookmarks.all
+  end
+
+  # GET /api/v1/bookmarks/1
+  def show; end
+
+  # POST /api/v1/bookmarks
+  def create
+    # TODO: find_or_create_by has a race condition and I don't have unique
+    # indexes on these yet as there is some cleanup around the uri string that
+    # I want to do first
+    webpage = Webpage.find_or_create_by(uri_string: params.dig(:data, :attributes, :url))
+    @bookmark = webpage.bookmarks.find_or_initialize_by(user: current_user)
+
+    if @bookmark.new_record?
+      @bookmark.update bookmark_params
+      if @bookmark.save
+        render :show, status: :created, location: @bookmark
+      else
+        render json: @bookmark.errors, status: :unprocessable_entity
+      end
+    else
+      render :show, status: :created, location: @bookmark
```

Comment:
should this be `status: :already_reported` or something else besides `:created`

Reply:
There isn't a good code that reports on upsert operations like this, that I've found, and returning a 302 Found a) didn't return a body when I tried it b) Still isn't the best code